python_coreml_stable_diffusion/torch2coreml.py

```diff
@@ -598,23 +630,59 @@ def convert_unet(pipe, args):
                           (batch_size)).to(torch.float32)),
             ("encoder_hidden_states", torch.rand(*encoder_hidden_states_shape))
         ])
-        sample_unet_inputs_spec = {
-            k: (v.shape, v.dtype)
-            for k, v in sample_unet_inputs.items()
-        }
-        logger.info(f"Sample inputs spec: {sample_unet_inputs_spec}")
-
-        # Initialize reference unet
-        reference_unet = unet.UNet2DConditionModel(**pipe.unet.config).eval()
-        load_state_dict_summary = reference_unet.load_state_dict(
-            pipe.unet.state_dict())
 
         # Prepare inputs
         baseline_sample_unet_inputs = deepcopy(sample_unet_inputs)
         baseline_sample_unet_inputs[
             "encoder_hidden_states"] = baseline_sample_unet_inputs[
                 "encoder_hidden_states"].squeeze(2).transpose(1, 2)
 
+        # Initialize reference unet
+        reference_unet = unet.UNet2DConditionModel(**pipe.unet.config).eval()
+        load_state_dict_summary = reference_unet.load_state_dict(
+            pipe.unet.state_dict())
+
+        if args.unet_support_controlnet:
```

Comment:
Is there any situation where `args.unet_support_controlnet` should be false when at least one `--convert-controlnet` argument is provided? Would it make sense to make `args.unet_support_controlnet` dependent on `--convert-controlnet` being provided?

Reply:
I assumed a situation like that we convert a model of Stable Diffusion to be compatible with ControlNet models which have already been converted along with another Stable Diffusion model. In that case, we wouldn't need to convert controlNet itself but making UNet support it would be required.

Reply:
Understood! My understanding is that ControlNet models are tethered to specific Stable Diffusion versions. Could you let me know if that is an accurate assumption?

Reply:
Either way, I think it should be fine if you would like to keep `args.unet_support_controlnet`.

Reply:
> ControlNet models are tethered to specific Stable Diffusion versions.

Yes, I think that's right. But controlnet, which was trained on a certain version of original stable diffusion, also looks working well when used for fine-tuned models derived from original one.
For example, https://huggingface.co/lllyasviel/sd-controlnet-mlsd was trained on SD 1.5 can be used for fine-tuned models derived from SD 1.5 as well.